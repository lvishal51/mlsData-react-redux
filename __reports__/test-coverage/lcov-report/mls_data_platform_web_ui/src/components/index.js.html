<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for mls_data_platform_web_ui/src/components/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">mls_data_platform_web_ui/src/components</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/0</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">export { default as ZapInput } from './zap-input/ZapInput';
export { default as ZapHeader } from './zap-header/ZapHeader';
export { default as ZapFooter } from './zap-footer/ZapFooter';
export { default as ZapPanel } from './zap-panel/ZapPanel';
export { default as MappedUnmapped } from './zap-mapping/MappedUnmapped';
export { default as CustomPagination } from './zap-pagination/CustomPagination';
export { default as MappedFields } from './zap-mappedFields/MappedFields';
export { default as ContextDataPanel } from './zap-context-panel/ContextDataPanel';
export { default as AutoSuggestion } from './zap-autoSuggestion/AutoSuggestion';
export { default as ZapTable } from './zap-table/ZapTable';
export { default as AutoComplete } from './zap-autoComplete/AutoComplete';
export { default as ZapSideBar } from './zap-sideBar/zapSideBar';
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Nov 29 2018 15:48:26 GMT+0530 (IST)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
