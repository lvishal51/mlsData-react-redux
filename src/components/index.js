export { default as ZapInput } from './zap-input/ZapInput';
export { default as ZapHeader } from './zap-header/ZapHeader';
export { default as ZapFooter } from './zap-footer/ZapFooter';
export { default as ZapPanel } from './zap-panel/ZapPanel';
export { default as MappedUnmapped } from './zap-mapping/MappedUnmapped';
export { default as CustomPagination } from './zap-pagination/CustomPagination';
export { default as MappedFields } from './zap-mappedFields/MappedFields';
export { default as ContextDataPanel } from './zap-context-panel/ContextDataPanel';
export { default as AutoSuggestion } from './zap-autoSuggestion/AutoSuggestion';
export { default as ZapPreview } from './zap-preview/ZapPreview';
export { default as AutoComplete } from './zap-autoComplete/AutoComplete';
export { default as CustomCalendar } from './zap-calendar/CustomCalendar';
export { ZapHeader as ZapHeaderComp } from './zap-header/ZapHeader';
export { ZapPanel as ZapPanelComp } from './zap-panel/ZapPanel';
export { ContextDataPanel as ContextDataPanelComp} from './zap-context-panel/ContextDataPanel';
export { AutoComplete as AutoCompleteComp } from './zap-autoComplete/AutoComplete';
export { MappedFields as MappedFieldsComp} from './zap-mappedFields/MappedFields';
