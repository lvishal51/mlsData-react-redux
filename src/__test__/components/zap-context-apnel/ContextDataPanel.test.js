import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { ContextDataPanelComp as ContextDataPanel } from 'components';
import { AutoSuggestion } from 'components'

const contextVariablesData = [
  {
    "id": 24,
    "mlsInfoId": 137,
    "type": {
      "id": 2,
      "type": "manually completed",
      "parentType": null,
      "lastUpdateTs": "2018-10-04T00:00:00.000Z",
      "createTs": "2018-10-04T00:00:00.000Z"
    },
    "key": "PRICE_FIELD",
    "value": "ListPrice",
    "elasticSearchIndexId": null,
    "hierarchyInfoId": 0,
    "lastUpdateTs": "2018-10-18T00:00:00.000Z",
    "createTs": "2018-10-18T00:00:00.000Z"
  },
  {
    "id": 22,
    "mlsInfoId": 137,
    "type": {
      "id": 2,
      "type": "manually completed",
      "parentType": null,
      "lastUpdateTs": "2018-10-04T00:00:00.000Z",
      "createTs": "2018-10-04T00:00:00.000Z"
    },
    "key": "STATUS_ACTIVE",
    "value": "200004325490",
    "elasticSearchIndexId": null,
    "hierarchyInfoId": 0,
    "lastUpdateTs": "2018-10-18T00:00:00.000Z",
    "createTs": "2018-10-18T00:00:00.000Z"
  },
  {
    "id": 23,
    "mlsInfoId": 137,
    "type": {
      "id": 2,
      "type": "manually completed",
      "parentType": null,
      "lastUpdateTs": "2018-10-04T00:00:00.000Z",
      "createTs": "2018-10-04T00:00:00.000Z"
    },
    "key": "STATUS_PENDING",
    "value": "200004324452",
    "elasticSearchIndexId": null,
    "hierarchyInfoId": 0,
    "lastUpdateTs": "2018-10-18T00:00:00.000Z",
    "createTs": "2018-10-18T00:00:00.000Z"
  },
];

const selectedEntryList = [
  {
    selectedKey: 'Resource',
    selectedValue: 'Property',
    selectedValueId: 'jbhjks7838',
    parentId: 1,
  },
  {
    selectedKey: 'Class',
    selectedValue: 'A',
    selectedValueId: 'fhgvhbm7289',
    parentId: 2,
  },
  {
    selectedKey: 'LookupName',
    selectedValue: 'OpenHouse',
    selectedValueId: 'abc133',
    parentId: 3,
  }
]

const autoSuggestData = [
  {
  "resource": "Property",
  },
  {
  "resource": "OpenHouse",
  },
];

const MlsContextVariableAction = {
  deleteContextVariables: jest.fn(),
  getMlsContextVariables: jest.fn(),
  getMetadataHierarchy: jest.fn(),
  addNewContextVariable: jest.fn(),
  getSuggestResource: jest.fn(),
  getSuggestClasses: jest.fn(),
  getSuggestSystemTableName: jest.fn(),
  getSuggestLookupName: jest.fn(),
  getSuggestLookupValue: jest.fn(),
  toogleContextVariableModal: jest.fn(),
  updateContextVariable: jest.fn(),
 };

const inputData = {
    contextVariablesData: contextVariablesData,
    MlsDeleteContextData: {
      isResponseSuccess: false
    },
    MlsContextVariables : {
      isResponseSuccess : false
    },
    MlsConfig : {
      isResponseSuccess : false
    },
    MlsResource: {
      isResponseSuccess: false
    },
    MlsClass : {
      isResponseSuccess: false
    },
    MlsSystemTable: {
      isResponseSuccess: false
    },
    MlsLookupName : {
      isResponseSuccess: false
    },
    MlsLookupValue: {
      isResponseSuccess: false
    },
    MlsHierarchy : {
      isResponseSuccess: false
    },
    MlsConfigData : [{
      source: 'FLEXMLS'
    }],
    MlsContextVariableAction
};


const inputState = {
  contextVariablesData: inputData.contextVariablesData,
  checkedContextVariables: inputData.contextVariablesData,
  formToggle: true,
  showAutoSuggestion: true,
  isOpenNewContextModal: true,
}

let wrapper;

beforeEach(() => {
  wrapper = shallow(<ContextDataPanel  {...inputData} />);
});


describe('Zap Context Variable Panel component', () => {

  it('should render correctly.', () => {
    wrapper.setProps({MlsContextVariables: { isResponseSuccess: true}});
    wrapper.setState({
      contextVariablesData: inputData.contextVariablesData,
      checkedContextVariables: inputData.contextVariablesData,
      formToggle: true,
    });
    wrapper.instance().forceUpdate();
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('checkboxClick function should be triggered correctly', () => {
    wrapper.setProps({MlsResource: { isResponseSuccess: true}});
    wrapper.setState({
      contextVariablesData: inputData.contextVariablesData,
      checkedContextVariables: inputData.contextVariablesData
    });
    const event = { preventDefault() {}, target: inputData[0] };
    const spy = jest.spyOn(wrapper.instance(), "checkboxClick");
    wrapper.instance().forceUpdate();
    wrapper.find(".checkbox_0").simulate('click', contextVariablesData[0]);
    expect(spy).toBeCalled();
  });

  it('deleteContextData function should be triggered correctly', () => {
    wrapper.setProps({MlsClass: { isResponseSuccess: true}});
    wrapper.setState(
      {
        contextVariablesData: inputData.contextVariablesData,
        checkedContextVariables: inputData.contextVariablesData,
        formToggle: true
      });
    const event = { preventDefault() {}, target: inputData[0] };
    const spy = jest.spyOn(wrapper.instance(), "deleteContextData");
    wrapper.instance().forceUpdate();
    wrapper.find(".delete").simulate('click', event);
    expect(spy).toBeCalled();
    expect(MlsContextVariableAction.deleteContextVariables).toBeCalled();
  });

  it('toggleButton function should be triggered correctly', () => {
    wrapper.setProps({MlsSystemTable: { isResponseSuccess: true}});
    wrapper.setState({ formToggle: true});
    const event = { preventDefault() {}, target: inputData[0] };
    const spy = jest.spyOn(wrapper.instance(), "toggleButton");
    wrapper.instance().forceUpdate();
    wrapper.find("#toggler").simulate('click', event);
    expect(spy).toBeCalled();
  });

  it('addNewContextToggle function should be triggered correctly', () => {
    wrapper.setProps({MlsLookupName: { isResponseSuccess: true}});
    wrapper.setState(
      {
        contextVariablesData: inputData.contextVariablesData,
        checkedContextVariables: inputData.contextVariablesData,
        formToggle: true
      });
    const event = { preventDefault() {}, target: inputData[0] };
    const spy = jest.spyOn(wrapper.instance(), "addNewContextToggle");
    wrapper.instance().forceUpdate();
    wrapper.find(".addMore").simulate('click', event);
    expect(spy).toBeCalled();
  });

  it('onInputChange function should be triggered correctly', () => {
    wrapper.setProps({MlsLookupValue: { isResponseSuccess: true}});
    wrapper.setState({...inputState});
    const event = { preventDefault() {}, target: { value: 'A' } };
    const spy = jest.spyOn(wrapper.instance(), "onInputChange");
    wrapper.instance().forceUpdate();
    wrapper.find("#contextVariableName").simulate('change', event);
    expect(spy).toBeCalled();
  });

  it('onInputChange function should be triggered correctly', () => {
    wrapper.setProps({MlsHierarchy: { isResponseSuccess: true}});
    wrapper.setState({
      ...inputState,
      contextVariableName: 'CLASS_AGENT',
      contextVariableValue: 'class_agent',
      mlsHierarchyType: 'MD'
    });
    const event = { preventDefault() {}, target: { value: 'A' } };
    const spy = jest.spyOn(wrapper.instance(), "onInputChange");
    wrapper.instance().forceUpdate();
    wrapper.find("#contextVariableName").simulate('change', event);
    expect(spy).toBeCalled();
  });

  it('onChangeHierarchyType function should be triggered correctly', () => {
    wrapper.setState({...inputState});
    const event = { preventDefault() {}, target: { value: 'A' } };
    const spy = jest.spyOn(wrapper.instance(), "onChangeHierarchyType");
    wrapper.instance().forceUpdate();
    wrapper.find("#MlsHierarchyType").simulate('change', event);
    expect(spy).toBeCalled();
  });

  it('onChangeHierarchyType function should be triggered correctly', () => {
    wrapper.setState({...inputState});
    const event = { preventDefault() {}, target: { value: 'MlsHierarchyType' } };
    const spy = jest.spyOn(wrapper.instance(), "onChangeHierarchyType");
    wrapper.instance().forceUpdate();
    wrapper.find("#MlsHierarchyType").simulate('change', event);
    expect(spy).toBeCalled();
  });

  it('_onAddNewContextVariable function should be triggered correctly for adding new variable', () => {
    wrapper.setState({
      ...inputState,
      disableAddNewContextButton: false
    });
    const event = { preventDefault() {}, target: { value: 'A' } };
    const spy = jest.spyOn(wrapper.instance(), "_onAddNewContextVariable");
    wrapper.instance().forceUpdate();
    wrapper.find("Button").at(1).simulate('click', event);
    expect(spy).toBeCalled();
  });

  it('_onAddNewContextVariable function should be triggered correctly for editing existing variable with different key', () => {
    jest.useFakeTimers();
    wrapper.setState({
      ...inputState,
      disableAddNewContextButton: false,
      isAdd: false,
      prevContextVariableName: 'abc',
      contextVariableName: 'bcd'
    });
    const event = { preventDefault() {}, target: { value: 'A' } };
    const spy = jest.spyOn(wrapper.instance(), "_onAddNewContextVariable");
    wrapper.instance().forceUpdate();
    wrapper.find("Button").at(1).simulate('click', event);
    expect(spy).toBeCalled();
    jest.runAllTimers();
  });

  it('_onAddNewContextVariable function should be triggered correctly for editing existing variable with same key', () => {
    wrapper.setState({
      ...inputState,
      disableAddNewContextButton: false,
      isAdd: false,
      prevContextVariableName: 'abc',
      contextVariableName: 'abc'
    });
    const event = { preventDefault() {}, target: { value: 'A' } };
    const spy = jest.spyOn(wrapper.instance(), "_onAddNewContextVariable");
    wrapper.instance().forceUpdate();
    wrapper.find("Button").at(1).simulate('click', event);
    expect(spy).toBeCalled();
  });

  it('onEditContextData function should be triggered correctly', () => {
    wrapper.setState({
      ...inputState,
      disableAddNewContextButton: false,
      isAdd: false,
      checkedContextVariables: inputData.contextVariablesData
    });
    const event = { preventDefault() {}, target: { value: 'A' } };
    const spy = jest.spyOn(wrapper.instance(), "onEditContextData");
    wrapper.instance().forceUpdate();
    wrapper.find(".edit").simulate('click', event);
    expect(spy).toBeCalled();
  });

  it('handleInputChange function should be triggered correctly in case of Resource', () => {
    wrapper.setState({
      ...inputState,
      autoSuggestData
    });
    const spy = jest.spyOn(wrapper.instance(), "handleInputChange");
    wrapper.instance().forceUpdate();
    wrapper.find(AutoSuggestion).prop('handleInputChange')('Resource', '', selectedEntryList);
    expect(spy).toBeCalled();
  });

  it('handleInputChange function should be triggered correctly in case of Class', () => {
    wrapper.setState({
      ...inputState,
      autoSuggestData
    });
    const spy = jest.spyOn(wrapper.instance(), "handleInputChange");
    wrapper.instance().forceUpdate();
    wrapper.find(AutoSuggestion).prop('handleInputChange')('Class', '', selectedEntryList);
    expect(spy).toBeCalled();
    wrapper.find(AutoSuggestion).prop('handleInputChange')('Class', '', {});
    expect(spy).toBeCalled();
  });

  it('handleInputChange function should be triggered correctly in case of TableSystemName', () => {
    wrapper.setState({
      ...inputState,
      autoSuggestData
    });
    const spy = jest.spyOn(wrapper.instance(), "handleInputChange");
    wrapper.instance().forceUpdate();
    wrapper.find(AutoSuggestion).prop('handleInputChange')('TableSystemName', '', selectedEntryList);
    expect(spy).toBeCalled();
    wrapper.find(AutoSuggestion).prop('handleInputChange')('TableSystemName', '', {});
    expect(spy).toBeCalled();
  });

  it('handleInputChange function should be triggered correctly in case of LookupName', () => {
    wrapper.setState({
      ...inputState,
      autoSuggestData
    });
    const spy = jest.spyOn(wrapper.instance(), "handleInputChange");
    wrapper.instance().forceUpdate();
    wrapper.find(AutoSuggestion).prop('handleInputChange')('LookupName', '', selectedEntryList);
    expect(spy).toBeCalled();
    wrapper.find(AutoSuggestion).prop('handleInputChange')('LookupName', '', {});
    expect(spy).toBeCalled();
  });

  it('handleInputChange function should be triggered correctly in case of LookupValue', () => {
    wrapper.setState({
      ...inputState,
      autoSuggestData
    });
    const spy = jest.spyOn(wrapper.instance(), "handleInputChange");
    wrapper.instance().forceUpdate();
    wrapper.find(AutoSuggestion).prop('handleInputChange')('LookupValue', '', selectedEntryList);
    expect(spy).toBeCalled();
    wrapper.find(AutoSuggestion).prop('handleInputChange')('LookupValue', '', {});
    expect(spy).toBeCalled();
  });

  it('handleInputChange function should be triggered correctly in case of default', () => {
    wrapper.setState({
      ...inputState,
      autoSuggestData
    });
    const spy = jest.spyOn(wrapper.instance(), "handleInputChange");
    wrapper.instance().forceUpdate();
    wrapper.find(AutoSuggestion).prop('handleInputChange')('LookupValue', inputData.MlsConfigData, selectedEntryList);
    expect(spy).toBeCalled();
    wrapper.find(AutoSuggestion).prop('handleInputChange')('', inputData.MlsConfigData, {});
    expect(spy).toBeCalled();
  });

  it('updateQuery function should be triggered correctly', () => {
    wrapper.setState({
      ...inputState,
      autoSuggestData
    });
    const spy = jest.spyOn(wrapper.instance(), "updateQuery");
    wrapper.instance().forceUpdate();
    wrapper.find(AutoSuggestion).prop('onUpdateValueToField')(selectedEntryList, true, true);
    expect(spy).toBeCalled();
    wrapper.find(AutoSuggestion).prop('onUpdateValueToField')({});
    expect(spy).toBeCalled();
  });

  it('updateQuery function should be triggered correctly', () => {
    wrapper.setState({
      ...inputState,
      autoSuggestData
    });
    const spy = jest.spyOn(wrapper.instance(), "updateQuery");
    wrapper.instance().forceUpdate();
    wrapper.find(AutoSuggestion).prop('onUpdateValueToField')(selectedEntryList, true, false);
    expect(spy).toBeCalled();
    wrapper.find(AutoSuggestion).prop('onUpdateValueToField')({});
    expect(spy).toBeCalled();
  });

  it('onSearchText function should be triggered correctly for searching in context variable panel with value', () => {
    wrapper.setProps({
      MlsContextVariablesData: inputData.contextVariablesData
    });
    const event = { target: { value: 'A' } };
    const spy = jest.spyOn(wrapper.instance(), "onSearchText");
    wrapper.instance().forceUpdate();
    wrapper.find(".contextSearch").simulate('change', event);
    expect(spy).toBeCalled();
  });

  it('onSearchText function should be triggered correctly for searching in context variable panel with no value', () => {
    wrapper.setProps({
      MlsContextVariablesData: inputData.contextVariablesData
    });
    const event = { target: { value: '' } };
    const spy = jest.spyOn(wrapper.instance(), "onSearchText");
    wrapper.instance().forceUpdate();
    wrapper.find(".contextSearch").simulate('change', event);
    expect(spy).toBeCalled();
  });
});
