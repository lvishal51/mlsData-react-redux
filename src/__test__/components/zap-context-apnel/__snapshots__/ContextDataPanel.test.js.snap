// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Zap Context Variable Panel component should render correctly. 1`] = `
<div
  className="contextPanelContainer"
>
  <Button
    className="fa fa-chevron-circle-right hideIcon"
    color="light"
    id="toggler"
    onClick={[Function]}
    tag="button"
  />
  <UncontrolledCollapse
    toggleEvents={
      Array [
        "touchstart",
        "click",
      ]
    }
    toggler="#toggler"
  >
    <Card
      tag="div"
    >
      <CardHeader
        className="textAlignment"
        tag="div"
      >
        Context Data
      </CardHeader>
      <CardBody
        tag="div"
      >
        <div>
          <FormGroup
            tag="div"
          >
            <Input
              className="contextSearch"
              name="search"
              onChange={[Function]}
              placeholder="Search"
              type="search"
              value=""
            />
          </FormGroup>
          <div
            className="checkboxList"
          >
            <div
              className="contextVariablesData"
            >
              <FormGroup
                className="checkboxRow"
                key="24"
                tag="div"
              >
                <Input
                  className="checkbox_0"
                  id="PRICE_FIELD"
                  name="PRICE_FIELD"
                  onClick={[Function]}
                  type="checkbox"
                  value="PRICE_FIELD = ListPrice"
                />
                <Label
                  for="PRICE_FIELD"
                  tag="label"
                  widths={
                    Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                    ]
                  }
                >
                  PRICE_FIELD = ListPrice
                </Label>
              </FormGroup>
              <FormGroup
                className="checkboxRow"
                key="22"
                tag="div"
              >
                <Input
                  className="checkbox_1"
                  id="STATUS_ACTIVE"
                  name="STATUS_ACTIVE"
                  onClick={[Function]}
                  type="checkbox"
                  value="STATUS_ACTIVE = 200004325490"
                />
                <Label
                  for="STATUS_ACTIVE"
                  tag="label"
                  widths={
                    Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                    ]
                  }
                >
                  STATUS_ACTIVE = 200004325490
                </Label>
              </FormGroup>
              <FormGroup
                className="checkboxRow"
                key="23"
                tag="div"
              >
                <Input
                  className="checkbox_2"
                  id="STATUS_PENDING"
                  name="STATUS_PENDING"
                  onClick={[Function]}
                  type="checkbox"
                  value="STATUS_PENDING = 200004324452"
                />
                <Label
                  for="STATUS_PENDING"
                  tag="label"
                  widths={
                    Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                    ]
                  }
                >
                  STATUS_PENDING = 200004324452
                </Label>
              </FormGroup>
            </div>
          </div>
        </div>
      </CardBody>
      <CardFooter
        className="row verticalDivider footerButton"
        tag="div"
      >
        <div
          className="col-5 addMore"
          onClick={[Function]}
        >
          <i
            className="fa fa-plus footerIcon"
            color="primary"
          />
          <Label
            tag="label"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            Add More
          </Label>
        </div>
        <div
          className="col-3 edit"
          onClick={[Function]}
        >
          <i
            className="fa fa-pencil footerIcon"
            color="primary"
          />
          <Label
            tag="label"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            Edit
          </Label>
        </div>
        <div
          className="col-4 delete"
          disabled={false}
          onClick={[Function]}
        >
          <i
            className="fa fa-trash footerIcon"
            color="primary"
          />
          <Label
            tag="label"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            Delete
          </Label>
        </div>
      </CardFooter>
    </Card>
  </UncontrolledCollapse>
  <Modal
    autoFocus={true}
    backdrop={true}
    backdropTransition={
      Object {
        "mountOnEnter": true,
        "timeout": 150,
      }
    }
    centered={false}
    className="contextModal"
    fade={true}
    isOpen={false}
    keyboard={true}
    modalTransition={
      Object {
        "timeout": 300,
      }
    }
    onClosed={[Function]}
    onOpened={[Function]}
    role="dialog"
    zIndex={1050}
  >
    <ModalHeader
      charCode={215}
      closeAriaLabel="Close"
      tag="h5"
      toggle={[Function]}
      wrapTag="div"
    >
      Add New
       Context Variable
    </ModalHeader>
    <ModalBody
      tag="div"
    >
      <FormGroup
        tag="div"
      >
        <Label
          for="MlsHierarchyType"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          Metadata Type
        </Label>
        <Input
          id="MlsHierarchyType"
          name="MlsHierarchyType"
          onChange={[Function]}
          type="select"
          value="MlsHierarchyType"
        >
          <option
            value="MlsHierarchyType"
          >
            Select Metadata Type
          </option>
          <option
            value="MD"
          >
            Metadata
          </option>
          <option
            value="MDL"
          >
            Metadata Lookup
          </option>
          <option
            value="Literal"
          >
            Literal
          </option>
        </Input>
      </FormGroup>
      <FormGroup
        tag="div"
      >
        <Label
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          Context Variable Name
        </Label>
        <Input
          id="contextVariableName"
          name="contextVariableName"
          onChange={[Function]}
          type="text"
          value=""
        />
      </FormGroup>
      <FormGroup
        tag="div"
      >
        <Label
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          Context Variable Value
        </Label>
        <Input
          disabled={true}
          id="contextVariableValue"
          name="contextVariableValue"
          onChange={[Function]}
          type="text"
          value=""
        />
      </FormGroup>
    </ModalBody>
    <ModalFooter
      tag="div"
    >
      <Button
        color="primary"
        disabled={true}
        onClick={[Function]}
        tag="button"
      >
        Save
      </Button>
    </ModalFooter>
  </Modal>
</div>
`;
